builder/common/step_security_group.go

```diff
@@ -136,10 +137,39 @@ func (s *StepSecurityGroup) Run(ctx context.Context, state multistep.StateBag) m
 
 	log.Printf("[DEBUG] Found security group %s", s.createdGroupId)
 
+	temporarySGSourceCidrs := s.TemporarySGSourceCidrs
+	if len(temporarySGSourceCidrs) == 0 && s.TemporarySGSourcePublicIp {
+		ui.Say("Checking current host's public IP...")
+
+		ip, err := CheckPublicIp()
+		if err != nil {
+			ui.Error(err.Error())
+			state.Put("error", err)
+			return multistep.ActionHalt
+		}
+
+		// ensure 0.0.0.0 isn't used to configure the SG
+		if ip.IsUnspecified() {
+			err := fmt.Errorf("Current host's public IP is unspecified: %s", ip)
+			ui.Error(err.Error())
+			state.Put("error", err)
+			return multistep.ActionHalt
+		}
+
+		ui.Say(fmt.Sprintf("Current host's public IP: %s", ip))
+
+		bits := 128
+		if tmp := ip.To4(); tmp != nil {
```

Comment:
This is somewhat misleading here, and to be clear this is not your fault, it's mostly on `ip.To4()` that attempts to parse the ip as V6 (i.e. 4 bytes) instead of v4, but may I suggest adding a little comment to clarify this bit of code?

```suggestion
		// ip.To4 attempts to parse the IP as IPv6, if this fails, we fallback to IPv4.
		if tmp := ip.To4(); tmp != nil {
```

Reply:
At this point in code I do not know if the IP address I received from AWS is v4 or v6.
All I know is that it is one of them.
 To format a correct CIDR I must choose the correct bitmask according to the address type.
`ip.To4()` returns a value if the address is v4, or `nil` otherwise (which means it is v6). So this is my test for choosing the correct bitmask.

> `// ip.To4 attempts to parse the IP as IPv6, if this fails, we fallback to IPv4.`

I am not sure I understand this comment, cause `To4` checks if the address is v4, it doesn't try to parse it as v6.

Did you mean `// ip.To4() attempts to parse the IP as IPv4, if this fails, we fallback to IPv6.` ?

Reply:
@lbajolet-hashicorp I have added a comment which I think clarifies the code.
Please tell me if that is OK.

Reply:
Oh you're right, I misunderstood the function as attempting to convert to IPv6, but it is indeed IPv4, sorry for that.
The corrected comment looks good to me